### ISSUE TITLE
Cannot read from VOTable as StringIO-wrapped string

### ISSUE THREAD

**Issue by mjgraham** [atpy/atpy#26]
```
>>> import atpy
>>> import StringIO
>>> x = open('test.vot').read()
>>> t = atpy.Table(StringIO.StringIO(x), type='vo')
/Library/Frameworks/Python.framework/Versions/6.0.0/lib/python2.6/site-packages/vo/voexceptions.py:62: UserWarning: None:4:0: W03: Implictly generating an ID from a name 'query results' -> 'query_results'
  warn(str(warning))
/Library/Frameworks/Python.framework/Versions/6.0.0/lib/python2.6/site-packages/vo/voexceptions.py:62: UserWarning: None:11:5: W06: Invalid UCD 'stat.error;phot.mag': Primary word 'phot.mag' is not valid as a secondary word
  warn(str(warning))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Library/Frameworks/Python.framework/Versions/6.0.0/lib/python2.6/site-packages/atpy/basetable.py", line 167, in __init__
    self.read(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/6.0.0/lib/python2.6/site-packages/atpy/basetable.py", line 213, in read
    atpy._readers[table_type](self, *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/6.0.0/lib/python2.6/site-packages/atpy/votable.py", line 86, in read
    votable = parse(filename, pedantic=pedantic)
  File "/Library/Frameworks/Python.framework/Versions/6.0.0/lib/python2.6/site-packages/vo/table.py", line 103, in parse
    return tree.VOTableFile(config=config, pos=(1, 1)).parse(iterator, config)
  File "/Library/Frameworks/Python.framework/Versions/6.0.0/lib/python2.6/site-packages/vo/tree.py", line 2832, in parse
    for start, tag, data, pos in iterator:
ValueError: 1:0: no element found
>>> 
```

**Comment by astrofrog** [comment]
This is strange - could you send me the test.vot file to thomas.robitaille@gmail.com? Thanks!

**Comment by astrofrog** [comment]
I think this is not the latest git version, because according to the traceback, line 86 of votable.py is:

```
votable = parse(filename, pedantic=pedantic)
```

when it should be:

```
elif len(tables) == 0:
```

See <a href="https://github.com/atpy/atpy/blob/master/atpy/votable.py#L86">here</a>. Try removing the `atpy` directory in `site-packages`, then install the latest git version with:

```
git clone git://github.com/atpy/atpy.git
cd atpy
python setup.py install
```

Let me know whether this helps solve the issue.

**Comment by astrofrog** [comment]
With the latest ATpy, I do get an error, but it is different, and is due to the fact that ATpy converts the data into a numpy structured array, and there must be a small bug in there:

```
ValueError: output operand requires a reduction, but reduction is not enabled
```

However, it is reading correctly from the file object, you get the same error if reading from a file. I'll look into this error.

**Comment by astrofrog** [comment]
For info, I've narrowed down the issue to the fact that Numpy automatically seems to want to convert `n x 1` arrays into `n` arrays:

```
In [1]: import numpy as np

In [2]: np.dtype([('a','<f4',2)])
Out[2]: dtype([('a', '<f4', (2,))])

In [3]: np.dtype([('a','<f4',1)])
Out[3]: dtype([('a', '<f4')])
```

I'll check with Numpy devs if this is a bug, and in any case, I'll add a workaround!

**Comment by astrofrog** [comment]
I fixed the bug in 4a0dd8293082feba8711

Your example now works correctly:

```
In [1]: import atpy
im
In [2]: import StringIO

In [3]: x = open('test.vot').read()

In [4]: t = atpy.Table(StringIO.StringIO(x), type='vo')
/var/folders/8L/8L-R-5PbEh0ef6pKH-3E5k+++TI/-Tmp-/tmp6nkM7A:4:0: W03: Implictly generating an ID from a name 'query results' -> 'query_results'
/var/folders/8L/8L-R-5PbEh0ef6pKH-3E5k+++TI/-Tmp-/tmp6nkM7A:11:5: W06: Invalid UCD 'stat.error;phot.mag': Primary word 'phot.mag' is not valid as a secondary word
/var/folders/8L/8L-R-5PbEh0ef6pKH-3E5k+++TI/-Tmp-/tmp6nkM7A:4:0: W03: Implictly generating an ID from a name 'query results' -> 'query_results'
/var/folders/8L/8L-R-5PbEh0ef6pKH-3E5k+++TI/-Tmp-/tmp6nkM7A:11:5: W06: Invalid UCD 'stat.error;phot.mag': Primary word 'phot.mag' is not valid as a secondary word
```

You can turn off the warnings with `verbose=False`. I'll close the ticket once you confirm this works correctly for you too!
